src/ui/actions/timeline.ts

```diff
@@ -174,8 +174,18 @@ export function setZoomRegion(region: ZoomRegion): SetZoomRegionAction {
   return { type: "set_zoom", region };
 }
 
-export function seek(point: ExecutionPoint, time: number, hasFrames: boolean): UIThunkAction {
+export function seek(
+  point: ExecutionPoint,
+  time: number,
+  hasFrames: boolean,
+  pauseId?: PauseId
+): UIThunkAction {
   return () => {
-    ThreadFront.timeWarp(point, time, hasFrames);
+    const pause = pauseId !== undefined ? Pause.getById(pauseId) : undefined;
+    if (pause) {
+      ThreadFront.timeWarpToPause(pause);
+    } else {
+      ThreadFront.timeWarp(point, time, hasFrames);
```

Comment:
now that we're sharing redux actions, we can probably get rid of the other timeWarpy actions